### PR TITLE
ProjectResolver throws when resolving ambiguous project names

### PR DESCRIPTION
**Pull request by ChengTian**
parent https://github.com/aspnet/dnx/issues/1818

### CODE REVIEW

File: src/Microsoft.Framework.Runtime/ProjectResolver.cs [hunk 1]
```diff
@@ -37,14 +37,21 @@ public bool TryResolveProject(string name, out Project project)
...
+                return false;
             }
 
-            return false;
+            var candidates = _projects[name].Where(p => File.Exists(Path.Combine(p.FullPath, Project.ProjectFileName)));
+            if (candidates.Count() > 1)
+            {
+                var allCandidatePaths = string.Join(Environment.NewLine, candidates.Select(x => x.FullPath).OrderBy(x => x));
+                throw new InvalidOperationException(
+                    $"'{name}' is an ambiguous name resolved to following projects:{Environment.NewLine}{allCandidatePaths}");
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/ProjectResolver.cs]: `The project name {name} is ambiguous between the following projects`

File: src/Microsoft.Framework.Runtime/ProjectResolver.cs [hunk 2]
```diff
@@ -37,14 +37,21 @@ public bool TryResolveProject(string name, out Project project)
...
-            if (_projects.TryGetValue(name, out projectInfo))
+            if (!_projects.Contains(name))
             {
-                project = projectInfo.Project;
-                return project != null;
+                return false;
             }
 
-            return false;
+            var candidates = _projects[name].Where(p => File.Exists(Path.Combine(p.FullPath, Project.ProjectFileName)));
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/ProjectResolver.cs]: This code paths runs every time somebody asks for a project. Checking the existence is why ProjectInformation.Project is lazy and only ever runs once.

**Comment by ChengTian** [comment]
@davidfowl , thanks for the feedback, updated.

**Comment by ChengTian** [comment]
Just came up with an edge case where you have the following `global.json`:
```json
{
  "projects": ["src", "src/../src"]
}
```
This will make `ProjectResolver` throw, even you only have a single project `src/ProjectA`. And the error message can be very confusing:
```
The project name 'ProjectA' is ambiguous between the following projects:
<Root>/src/ProjectA
<Root>/src/ProjectA
```

Need to dedup. Working on it.

**Comment by ChengTian** [comment]
@davidfowl , added dedup of project path and added tests for the case above.

**Comment by davidfowl** [comment]
Good catch! :shipit:

**Pull request merged**